src/main.rs

```diff
@@ -508,6 +510,27 @@ fn run(args: Cli) -> Result<i32> {
 
         let mut request = request_builder.headers(headers).build()?;
 
+        if args.compress >= 1 && request.headers().get(CONTENT_ENCODING).is_none() {
+            let mut compressed = false;
+            if let Some(body) = request.body_mut() {
+                if let Some(body_bytes) = body.as_bytes() {
```

Comment:
Can we make this work for streaming requests as well (i.e. `xh -x : @file.txt`)? In that case it might be best to just compress unconditionally even with a single `-x` so that we don't have to buffer it.

(HTTPie does buffer the whole file in this case.)

Reply:
I think it's possible; the latest commit supports this. 😄

Reply:
Awesome, that gets us the same behavior as HTTPie. But we might want to diverge from HTTPie by streaming the body in this case instead of buffering it, so that it's possible to upload files that don't fit in RAM. That should be possible with [`ReqwestBody::new`](https://docs.rs/reqwest/0.12.12/reqwest/blocking/struct.Body.html#method.new).

(I also suggested this in https://github.com/httpie/cli/pull/1613 so we could wait to see what they think.)

Reply:
We can directly compress the stream without buffering it, which would make it different from httpie's implementation. 
Should we do it this way in this PR?